infra/core/storage/storage-account.bicep

```diff
@@ -25,13 +25,20 @@ param publicNetworkAccess string = 'Enabled'
 param sku object = { name: 'Standard_LRS' }
 @allowed([ 'None', 'AzureServices' ])
 param bypass string = 'AzureServices'
+param storageInfrastructureEncryption string = 'Disabled'
 
 var networkAcls = (publicNetworkAccess == 'Enabled') ? {
   bypass: bypass
   defaultAction: 'Allow'
 } : { defaultAction: 'Deny' }
 
-resource storage 'Microsoft.Storage/storageAccounts@2022-05-01' = {
+var encryption = (storageInfrastructureEncryption == 'Enabled') ? {
```

Comment:
If it's going to be mapped to this object anyway, would it be better to make it a boolean?

Reply:
I just looked up the avm module, which we'll eventually move to, and it uses a boolean:
https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/storage/storage-account#parameter-requireinfrastructureencryption